Source/JavaScriptCore/API/JSValue.mm

```diff
@@ -1261,10 +1262,9 @@ - (JSValue *)initWithValue:(JSValueRef)value inContext:(JSContext *)context
     // Step 1: find all valueWith<Foo>:inContext: class methods in JSValue.
     forEachMethodInClass(object_getClass([JSValue class]), ^(Method method){
         SEL selector = method_getName(method);
-        const char* name = sel_getName(selector);
-        size_t nameLength = strlen(name);
+        auto name = span(sel_getName(selector));
         // Check for valueWith<Foo>:context:
-        if (nameLength < valueWithXinContextLength || memcmp(name, "valueWith", 9) || memcmp(name + nameLength - 11, ":inContext:", 11))
+        if (name.size() < valueWithXinContextLength || !equalSpans(name.first(9), "valueWith"_span) || !equalSpans(name.last(11), ":inContext:"_span))
```

Comment:
Would nicer if the `9` and the `11` were computed at compile time or even runtime. Like a `hasPrefix` function.

Reply:
Ok, proposing:
```
template<typename T, std::size_t TExtent, typename U, std::size_t UExtent>
bool spanHasPrefix(std::span<T, TExtent> span, std::span<U, UExtent> prefix)
{
    static_assert(sizeof(T) == sizeof(U));
    static_assert(std::has_unique_object_representations_v<T>);
    static_assert(std::has_unique_object_representations_v<U>);
    if (span.size() < prefix.size())
        return false;
    return !memcmp(span.data(), prefix.data(), prefix.size_bytes());
}
```